### ISSUE TITLE
Fabric: Change in a JS view's `accessibilityValue` prop does not fire the appropriate UIA update event

### ISSUE THREAD

**Issue by jonthysell** [microsoft/react-native-windows#12361]
### Problem Description

In PR #12287, support for the `accessbilityValue` prop was added. However it seems like only the `TextInput` actually files an event (when the text changes). However other custom views which set (and change) that property won't fire an event to notify UIA.

I.e. ComponentBaseComponentView should have something like this in `updateAccessibilityProps`:

```
winrt::Microsoft::ReactNative::implementation::UpdateUiaProperty(
      m_uiaProvider, UIA_ValueValuePropertyId, oldViewProps.accessibilityValue.text, newViewProps.accessibilityValue.text);
```

### Steps To Reproduce

Code analysis

### CLI version

npx react-native -v

### Environment

```markdown
npx react-native info
```

### Target Platform Version

None

### Visual Studio Version

None

### Build Configuration

None

**Comment by vineethkuttan** [comment]
Fixed in #14858
